SmartDeviceLink-iOS/SmartDeviceLink/SDLLocalization.m

```diff
@@ -0,0 +1,198 @@
+// SDLLocalization.m
+//
+
+
+#import "SDLLocalization.h"
+
+/** Private definition extending setter for the properties. */
+@interface SDLLocalization ()
+
+@property (readwrite) NSString *language;
+
+@property (readwrite) NSString *region;
+
+@property (readwrite) NSBundle *defaultBundle;
+
+@property (readwrite) NSLocale *defaultLocale;
+
+@property (readwrite) NSBundle *fallbackBundle;
+
+@property (readwrite) NSLocale *fallbackLocale;
+
+@end
+
+@implementation SDLLocalization
+
+- (NSBundle *)bundle {
+    if ([self defaultBundle]) {
+        return [self defaultBundle];
+    } else if ([self fallbackBundle]) {
+        return [self fallbackBundle];
+    } else {
+        return nil;
+    }
+}
+
+- (NSLocale *)locale {
+    if ([self defaultLocale]) {
+        return [self defaultLocale];
+    } else if ([self fallbackLocale]) {
+        return [self fallbackLocale];
+    } else {
+        return nil;
+    }
+}
+
++ (instancetype)defaultLocalization {
+    static SDLLocalization *object = nil;
+    static dispatch_once_t token;
+
+    dispatch_once(&token, ^{
+        // get the locale identifier
+        NSString *localeIdentifier = [[NSLocale currentLocale] localeIdentifier];
```

Comment:
Both options are not covering what languages the app is capable of. I assume `currentLocale` does that and in addition it tells me the region which `preferred...` doesn't. It's necessary to load the correct bundle.

While testing this I have never seen the issue mentioned on StackOverflow. `currentLocale` was always returning the expected value. But I will have a look at it.

Regarding your EDIT. Language only does not fulfill the full localization feature. See other comment below to line 108.